src/tracks/Pattern.cpp

```diff
@@ -65,6 +65,39 @@ Pattern::Pattern( InstrumentTrack * _instrument_track ) :
 	m_steps( MidiTime::stepsPerTact() )
 {
 	setName( _instrument_track->name() );
+
+	// Resize this track to be the same as existing tracks in the BB
+	// Only if there already exists at least another track in the BB
+	const TrackContainer::TrackList & tracks = m_instrumentTrack->trackContainer()->tracks();
+	if(tracks.size() > 1)
+	{
+		// Get currentTCO index (current BB)
+		int currentTCO = m_instrumentTrack->getTCOs().indexOf(this);
+		// Get first InstrumentTrack
+		unsigned int trackID = 0;
+		for(; trackID < tracks.size(); ++trackID)
+		{
+			// skip track that is the same as the current Pattern
+			if(tracks.at(trackID) == static_cast<Track *>(m_instrumentTrack))
+			{
+				continue;
+			}
+			// Found existing InstrumentTrack, thus break
+			else if(tracks.at(trackID)->type() == Track::InstrumentTrack)
+			{
+				break;
+			}
+		}
+		// If trackID is out of bounds, then there is no valid InstrumentTrack
+		// If currentTCO is out of bounds, then a new BB was created and should
+		// have default size
+		if(trackID < tracks.size() && currentTCO < tracks.at(trackID)->numOfTCOs())
```

Comment:
`currentTCO` will be in bounds since the reference track comes before and is already initialized.